ddtrace/_trace/processor/__init__.py

```diff
@@ -136,7 +136,9 @@ def process_trace(self, trace: List[Span]) -> Optional[List[Span]]:
             root_ctx = chunk_root._context
 
             if self.apm_opt_out:
-                chunk_root.set_metric(MK_APM_ENABLED, 0)
+                for span in trace:
+                    if span._local_root_value is None:
+                        span.set_metric(MK_APM_ENABLED, 0)
```

Comment:
no it is not working because I detected multiple root spans at some occasion in that list. The assumption that there is only one root span is not true.